workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.test.ts

```diff
@@ -777,52 +777,132 @@ describe('CognitoUserManagementPlugin tests', () => {
         .on(AdminListGroupsForUserCommand)
         .resolves({ Groups: roles.map((role) => ({ GroupName: role })) });
 
-      const users = await plugin.listUsers();
+      const users = await plugin.listUsers({});
 
-      expect(users.length).toBe(0);
-      expect(users).toMatchObject([]);
+      expect(users.data.length).toBe(0);
+      expect(users).toMatchObject({ data: [] });
     });
 
     it('should populate missing values with empty strings', async () => {
       cognitoMock.on(ListUsersCommand).resolves({ Users: [{ Username: userInfo.id }] });
       cognitoMock.on(AdminListGroupsForUserCommand).resolves({ Groups: roles.map(() => ({})) });
 
-      const users = await plugin.listUsers();
+      const users = await plugin.listUsers({});
 
-      expect(users.length).toBe(1);
-      expect(users).toMatchObject<User[]>([
-        { id: userInfo.id, firstName: '', lastName: '', email: '', status: Status.INACTIVE, roles: [] }
-      ]);
+      expect(users.data.length).toBe(1);
+      expect(users).toMatchObject({
+        data: [
+          { id: userInfo.id, firstName: '', lastName: '', email: '', status: Status.INACTIVE, roles: [] }
+        ]
+      });
+    });
+
+    it('should return a pagination token in response when Cognito returns PaginationToken', async () => {
+      cognitoMock.on(ListUsersCommand).resolves({
+        Users: [
+          {
+            Username: userInfo.id,
+            Attributes: [
+              { Name: 'given_name', Value: userInfo.firstName },
+              { Name: 'family_name', Value: userInfo.lastName },
+              { Name: 'email', Value: userInfo.email }
+            ],
+            Enabled: true
+          }
+        ],
+        PaginationToken:
+          'IkNBSVNxQUlJQVJLQkFnZ0RFdndCQUgwVkljdU9TWFc1TWhieDliN2xZZWRQK3BQVzJ2NzRKbXF6YysySzVycUFleUpBYmlJNklsQmhaMmx1WVhScGIyNURiMjUwYVc1MVlYUnBiMjVFVkU4aUxDSnVaWGgwUzJWNUlqb2lRVUZCUVVGQlFVRkRUakJNUVZGRlFsbFhaV1JyWWpCUVUzQnBVVU40Vm5CV1oxZEtTWGxpSzBWUk5VMUhWVE5oYVRaRGFsRjROR1pWZGpWc1ltMVpOMDVVU1RCT2VsVjZUMGRGZEUxVVFtdFplVEF3VDFSRk0weFVaekZPVkZWMFRVZEthazlYU20xWlYxSnFUMGRLYTA5M1BUMGlMQ0p3WVdkcGJtRjBhVzl1UkdWd2RHZ2lPakVzSW5CeVpYWnBiM1Z6VW1WeGRXVnpkRlJwYldVaU9qRTJPRFV3TnpReE5qSXdOelI5R2lDRnpmNlU4OExHVFlSVU9wdWxCRFQ4Qlh6ZGkyVk55MDVObFVPY2Irci9qQT09Ig=='
+      });
+      cognitoMock
+        .on(AdminListGroupsForUserCommand)
+        .resolves({ Groups: roles.map((role) => ({ GroupName: role })) });
+
+      const users = await plugin.listUsers({ pageSize: 1 });
+
+      expect(users.data.length).toBe(1);
+      expect(users.paginationToken).toBeDefined();
```

Comment:
nit: we should verify that the paginationToken is strictEqual to the value you mocked above

Reply:
The paginationToken returned won't equal the pagination token mocked above because we base64 encode the one returned by Cognito within the method (since the pagination token returned by Cognito can have API-URL illegal symbols).